molecule/eval/converge.yml

```diff
@@ -217,6 +217,50 @@
               "--- {'other': 'cute-table'}\n...\n",
           ]
 
+# EVAL FROM STRING (with retries)
+
+- name: 'Eval with retries'
+  hosts: cluster
+  become: true
+  become_user: root
+  gather_facts: false
+  tasks:
+    - name: Eval returns nil, err
+      import_role:
+        name: ansible-cartridge
+      ignore_errors: true
+      vars:
+        cartridge_scenario:
+          - eval
+        cartridge_eval_body: return nil, "I am horrible error"
+        cartridge_eval_with_retries: true
+        cartridge_eval_retries: 2
+        cartridge_eval_delay: 1
+
+    - name: 'Check eval failed'
+      assert:
+        msg: 'Eval should fail'
+        success_msg: 'Eval failed'
+        that: eval_res_register.failed
+      run_once: true
+
+    - name: Eval returns res
+      import_role:
+        name: ansible-cartridge
+      vars:
+        cartridge_scenario:
+          - eval
+        cartridge_eval_body: return "I am some res"
+        cartridge_eval_with_retries: true
+        cartridge_eval_retries: 2
+        cartridge_eval_delay: 1
+
+    - name: 'Check result'
+      assert:
+        msg: 'Received bad result'
+        success_msg: 'Received result is OK'
+        that: eval_res == ["I am some res"]
```

Comment:
eval_res_register.failed == false?

Reply:
If so, execution fails earlier. For negative cases tests `ignore_errors: true` and `eval_res_register.failed` check is used, but here we don't need it.